python/golem_task_api/client.py

```diff
@@ -186,21 +226,18 @@ class ProviderAppClient:
             service: TaskApiService,
             port: int = DEFAULT_PORT
     ) -> 'ProviderAppClient':
-        host, port = await service.start(f'provider {port}', port)
-        await wait_until_socket_open(host, port)
-        app_stub = ProviderAppStub(
-            Channel(host, port, loop=asyncio.get_event_loop())
-        )
-        return cls(service, app_stub)
+
+        async with service.started_ctx(f'provider {port}', port) as channel:
```

Comment:
But we don't gain anything by using the contextmanager here, do we? It'll only benefit if the constructor below throws which looks like won't happen.

Reply:
Well, I thought something might go wrong in the Stub constructor. But if you think that is not a risk, I might do away with the context manager as well.

Reply:
OK, looked now and it's all just assignments. I think I may remove the context manager.